src/appleseed/renderer/modeling/bsdf/bsdffactoryregistrar.cpp

```diff
@@ -86,6 +86,12 @@ BSDFFactoryRegistrar::BSDFFactoryRegistrar(const SearchPaths& search_paths)
     impl->register_factory(auto_release_ptr<FactoryType>(new KelemenBRDFFactory()));
     impl->register_factory(auto_release_ptr<FactoryType>(new LambertianBRDFFactory()));
     impl->register_factory(auto_release_ptr<FactoryType>(new MetalBRDFFactory()));
+    impl->register_factory(auto_release_ptr<FactoryType>(new MicrofacetGlossyBRDFFactory()));
+    impl->register_factory(auto_release_ptr<FactoryType>(new MicrofacetBlinnBRDFFactory()));
+    impl->register_factory(auto_release_ptr<FactoryType>(new MicrofacetLambertianBRDFFactory()));
+    impl->register_factory(auto_release_ptr<FactoryType>(new MicrofacetMetalBRDFFactory()));
+    impl->register_factory(auto_release_ptr<FactoryType>(new MicrofacetPlasticBRDFFactory()));
+    impl->register_factory(auto_release_ptr<FactoryType>(new MicrofacetSheenBRDFFactory()));
     impl->register_factory(auto_release_ptr<FactoryType>(new OrenNayarBRDFFactory()));
```

Comment:
Is there a reason we only got that selection of BRDFs and not all regular BRDFs as microfacet as well (e.g. the `OrenNayarBRDF`)?

Reply:
It was important for me to have metal and glossy wrappers.

The reason for other selection is that I think it is various enough. New microfacet wrappers can be added with steps described here: https://github.com/appleseedhq/appleseed/pull/2886#issue-467422364.

Metal, Plastic, Glossy, Blinn are microfacet based BRDFs.

Lambertian is simple diffuse based.

Sheen has its own base: https://disney-animation.s3.amazonaws.com/library/s2012_pbs_disney_brdf_notes_v2.pdf